### ISSUE TITLE
implement autonyms

### ISSUE THREAD

**Issue by mfrasca** [Ghini/ghini.desktop#457]
quoting (and paraphrasing) @RoDuth:
»if a subspecies, etc. is created of a species then the type species automatically gets a subsp. name that mirrors the species epithet. E.g. when they decided that the _Araucaria cunninghamii_ from PNG was a variety and named it _Araucaria cunninghamii_ var. _papuana_ then« ( _Araucaria cunninghamii_ automatically became a synonym of _Araucaria cunninghamii_ var. _cunninghamii_).

the other way, quoting (with minor added text) @Ejgouda:
»the moment _Aechmea minor_ f. _ubi-major_ becomes a synonym of an other taxon, _Aechmea minor_ f. _minor_ (its autonym) does not exist any longer (not needed), but in the database there are probably many references pointing to that record, like identifications etc. I solve this problem by considering _Aechmea minor_ f. _minor_ synonym to _Aechmea minor_ by making it synonym to the latter. (which amounts to changing the verse of the synonymy link between the two)«

it would be good to have support for this in the software.

in practice, when you add an infraspecific taxon to a species, the species gets an automatic authonym and this autonym becomes the accepted name for the taxon.  there is no extra author for it, as there is no publication other than the original species publication.
when you remove the last infraspecific taxon at a given rank from a species, its automatic autonym at that rank should be removed, or become a transparent synonym to the species.
I can imagine cases like knowing of the existence of infraspecific taxa for a species, but not having them in the collection, so that you would want to add the autonym manually, and that it stays in the database even in absence of siblings at the same rank.

**Comment by Ejgouda** [comment]
Op 18-10-18 om 16:08 schreef Mario Frasca:

> quoting (and paraphrasing) @RoDuth <https://github.com/RoDuth>:
> »if a subspecies, etc. is created of a species then the type species 
> automatically gets a subsp. name that mirrors the species epithet. 
> E.g. when they decided that the /Araucaria cunninghamii/ from PNG was 
> a variety and named it /Araucaria cunninghamii/ var. /papuana/ then« ( 
> /Araucaria cunninghamii/ automatically became a synonym of /Araucaria 
> cunninghamii/ var. /cunninghamii/).
>
> the other way, quoting (with minor added text) @Ejgouda 
> <https://github.com/Ejgouda>:
> »the moment /Aechmea minor/ f. /ubi-major/ becomes a synonym of an 
> other taxon, /Aechmea minor/ f. /minor/ (its autonym) does not exist 
> any longer (not needed), but in the database there are probably many 
> references pointing to that record, like identifications etc. I solve 
> this problem by considering /Aechmea minor/ f. /minor/ synonym to 
> /Aechmea minor/ by making it synonym to the latter. (which amounts to 
> changing the verse of the synonymy link between the two)«
>
> it would be good to have support for this in the software.
>
> in practice, when you add an infraspecific taxon to a species, the 
> species gets an automatic authonym and this autonym becomes the 
> accepted name for the taxon. there is no extra author for it, as there 
> is no publication other than the original species publication.
>
This is not the right way to do it. You can have a pool of accessions 
pointing to it and you do not know if there are accessions in between 
that are the new infraspecific taxon. Also when in a new identification 
when you can not decide which infraspecific taxon it is, you just 
identify it on species level.

**Comment by mfrasca** [comment]
Eric, I have a question for you, and your list gives the example I was
missing.

    Tillandsia ixioides Griseb.
    Tillandsia ixioides subsp. viridiflora (Rauh) Gouda
    Tillandsia ixioides var. occidentalis A.Cast.

how would you indicate the 'true' Tillandsia ixioides?  which autonym I
mean.

**Comment by Ejgouda** [comment]
Tillandsia ixioides Griseb. subsp. ixioides for the subsp.

Tillandsia ixioides Griseb. subsp. ixioides var. ixioides or Tillandsia 
ixioides Griseb. var. ixioides for the var.

Because infraspecific epithets should be unique within a species, you 
can leave subsp. ixioides out for the var. autonym, it is facultative.

Hope this answered your question

Eric



Op 19-10-18 om 13:19 schreef Mario Frasca:

**Comment by mfrasca** [comment]
when you say "infraspecific epithets should be unique within a species", you mean across ranks, don't you?  e.g.: once you have taken 'viridiflora' for a subspecies, the same 'viridiflora' cannot be assigned to a variety of the same species, is this correct?

**Comment by Ejgouda** [comment]
Yes, for example if you use the epithet viridiflora for an infraspecific 
taxon, whatever rank, you can not use it for an other rank

if you have Tillandsia tenuifolia var. nigrifolia forma alba, you can 
not ad a new for example Tillandsia tenuifolia var. tenuifolia forma 
alba, otherwise you could not leave out "var. tenuifolia" from the name. 
Tillandsia tenuifolia forma alba is equivalent with Tillandsia 
tenuifolia var. nigrifolia forma alba. On other ranks, it is also not 
possible to create for example Tillandsia tenuifolia subsp. alba. 
Although the rank in the name does make a difference, it is not allowed, 
so that you always know that Tillandsia tenuifolia  forma alba is 
synonym with Tillandsia tenuifolia subsp. alba (it is the same taxon).

Eric


Op 22-10-18 om 01:39 schreef Mario Frasca:

**Comment by mfrasca** [comment]
> if you use the epithet viridiflora for an infraspecific taxon, whatever rank, you can not use it for an other rank

but the "you can not use", is it a guarantee that it never happened, so we don't need to represent it in the database?  I'm thinking of homonyms… (same rank, same epithet, other publication)

so: I cannot enforce uniqueness in the database table, or I would rule out homonyms.

OTOH, ranks in the range Ordo-Subtribus use suffixes that guarantee against cross-rank name clashes, but I've also seen epithets that didn't reveal the rank.  

so: are cross-rank homonyms a possibility?

**Comment by Ejgouda** [comment]
Op 25-10-18 om 16:03 schreef Mario Frasca:
>
>     if you use the epithet viridiflora for an infraspecific taxon,
>     whatever rank, you can not use it for an other rank
>
> but the "you can not use", is it a guarantee that it never happened, 
> so we don't need to represent it in the database? I'm thinking of 
> homonyms… (same rank, same epithet, other publication)
>
Of course it is possible to publish an invalid or illegitimate name, so 
you do not rule it out in the database

**Comment by RoDuth** [comment]
> in practice, when you add an infraspecific taxon to a species, the species gets an automatic authonym and this autonym becomes the accepted name for the taxon. there is no extra author for it, as there is no publication other than the original species publication.

> This is not the right way to do it. You can have a pool of accessions pointing to it and you do not know if there are accessions in between that are the new infraspecific taxon. Also when in a new identification when you can not decide which infraspecific taxon it is, you just identify it on species level.

I think I am just repeating @Ejgouda here...

A word of warning about autonyms.  This must be a manual step, per accession and no forced use of nor inclusion of the autonym.  I noted autonyms not because I wanted ghini to fix them for me but because they were an example of when following the strict rules of taxonomy in a world which is filled with commercial nursery, hobbyist collectors and specialist breeders, etc. could come unstuck for the average user.

In a practical working living collection the taxonomy is often full of inconsistencies and disputes, etc.  As @Ejgouda  points out.
> Of course it is possible to publish an invalid or illegitimate name, so you do not rule it out in the database
